### PR TITLE
Adds FragmentBundler

### PR DESCRIPTION
**Pull request by felipecsl**
As proposed in [this gist](https://gist.github.com/felipecsl/45bc010713c0f543354e)

### CODE REVIEW

File: bundler/src/main/java/com/f2prateek/bundler/FragmentBundler.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    bundlerBuilder = new FragmentBundleBuilder<F>(this);
+  }
+
+  public static <F extends Fragment> FragmentBundleBuilder<F> make(F fragment) {
+    return new FragmentBundler<F>(fragment).bundle();
+  }
+
+  public static <F extends Fragment> FragmentBundleBuilder<F> make(Class<F> klass)
+      throws IllegalAccessException, InstantiationException {
+    return make(klass.newInstance());
```
Comment (f2prateek) [inline bundler/src/main/java/com/f2prateek/bundler/FragmentBundler.java]: I think we should just throw an unchecked `IllegalArgumentException` (or something more appropriate) if `newInstance` throws either `IllegalAccessException` or `InstantiationException`.

File: bundler/src/main/java/com/f2prateek/bundler/FragmentBundler.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
+package com.f2prateek.bundler;
+
+
+import android.app.Fragment;
+
+/**
+ * Set fragment arguments like a Pro.
+ * @param <F> The fragment to setup.
+ */
+public class FragmentBundler<F extends Fragment> {
```
Comment (f2prateek) [inline bundler/src/main/java/com/f2prateek/bundler/FragmentBundler.java:17]: I'm a bit confused by the purpose of this class, looks like all it does is expose the two factory methods that return `FragmentBundleBuilder`. So if clients can never create or access methods on this class, why not just nix it?

File: bundler/src/main/java/com/f2prateek/bundler/FragmentBundleBuilder.java [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+import java.io.Serializable;
+import java.util.ArrayList;
+
+/**
+ * A bundle builder for fragment arguments.
+ *
+ * @param <F> The fragment type you're bundling to.
+ */
+@SuppressWarnings("unchecked")
+class FragmentBundleBuilder<F extends Fragment> extends Bundler {
```
Comment (f2prateek) [inline bundler/src/main/java/com/f2prateek/bundler/FragmentBundleBuilder.java]: 1. How about just naming this `FragmentBundler`?
2. Does it need to extend `Bundler`? I think it makes more sense for this to wrap a `Bundler` instance internally that it simply delegates to.

File: bundler/src/main/java/com/f2prateek/bundler/FragmentBundleBuilder.java [hunk 4]
```diff
@@ -0,0 +1,419 @@
...
+
+  private final FragmentBundler<F> mFragmentBundler;
```
Comment (f2prateek) [inline bundler/src/main/java/com/f2prateek/bundler/FragmentBundleBuilder.java]: Nix the `m` prefix to stay consistent across the project.

**Comment by f2prateek** [comment]
Sweet, LGTM! Just some tiny nits.

File: bundler/src/main/java/com/f2prateek/bundler/FragmentBundler.java (hunk 1 above)
Comment (f2prateek) [inline bundler/src/main/java/com/f2prateek/bundler/FragmentBundler.java]: And of course, document that a no-args constructor is required.

Comment (felipecsl) [inline bundler/src/main/java/com/f2prateek/bundler/FragmentBundler.java]: Agreed

File: bundler/src/main/java/com/f2prateek/bundler/FragmentBundler.java (hunk 2 above)
Comment (felipecsl) [inline bundler/src/main/java/com/f2prateek/bundler/FragmentBundler.java:17]: I agree. Initially for some reason I had the idea that these needed to be 2 separate classes but looking again right now it doens't make sense and in fact it is confusing. I'll merge them together.

File: bundler/src/main/java/com/f2prateek/bundler/FragmentBundleBuilder.java (hunk 3 above)
Comment (felipecsl) [inline bundler/src/main/java/com/f2prateek/bundler/FragmentBundleBuilder.java]: In the first version, I had it wrap the `Bundler` but then you have pretty much all the Bundler code repeated here, it's a lot of duplication, plus this class feels like it is just a specialization of the Bundler that also knows how to pass that info into a Fragment, so I think it makes more sense to extend it.

**Comment by felipecsl** [comment]
PR updated based on earlier comments

File: bundler/src/main/java/com/f2prateek/bundler/FragmentBundleBuilder.java (hunk 3 above)
Comment (f2prateek) [inline bundler/src/main/java/com/f2prateek/bundler/FragmentBundleBuilder.java]: Hmm...that is a good point, but users wouldn't really ever use this class as a `Bundler`, and so the inheritance doesn't serve a purpose other than reducing code.

``` java
 public FragmentBundleBuilder<F> put(String key, boolean value) {
    return (FragmentBundleBuilder<F>) super.put(key, value);
}
```

``` java
 public FragmentBundleBuilder<F> put(String key, boolean value) {
   bundle.putBoolean(key, value);
   return this;
}
```

I don't think it's that much extra code we're saving, and we can also avoid the ugliness of the casting if we wrap a Bundler (or a `Bundle`).

Comment (felipecsl) [inline bundler/src/main/java/com/f2prateek/bundler/FragmentBundleBuilder.java]: You made a good point. Fixing

**Comment by felipecsl** [comment]
done!

**Comment by f2prateek** [comment]
Thanks!

**Comment by felipecsl** [comment]
:) please let me know once you release the next version with these changes, I'd like to use it in my current project!

**Comment by f2prateek** [comment]
Releasing it now, should be out in a couple of hours through Maven.

**Pull request merged**